plugin.video.dr.dk.bonanza/resources/lib/addon.py

```diff
@@ -0,0 +1,228 @@
+#
+#      Copyright (C) 2014 Tommy Winther, termehansen
+
+#
+#  https://github.com/xbmc-danish-addons/plugin.video.drnu#
+#
+#  This Program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2, or (at your option)
+#  any later version.
+#
+#  This Program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this Program; see the file LICENSE.txt.  If not, write to
+#  the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.
+#  http://www.gnu.org/copyleft/gpl.html
+#
+import os
+import re
+import sys
+import requests
+import requests_cache
+from html import unescape
+import buggalo
+
+import xbmc
+import xbmcgui
+import xbmcplugin
+import xbmcaddon
+import json
+
+try:
+    # python 2
+    from urlparse import parse_qsl
+except:
+    from urllib.parse import parse_qsl
+
+BASE_URL = 'http://www.dr.dk/bonanza/'
+tr = xbmcaddon.Addon().getLocalizedString
```

Comment:
You might want to reuse the same add-on() instance instead of creating new ones all the time